### PR TITLE
Implement automation

### PR DESCRIPTION
**Pull request by E-Kemp**

### CODE REVIEW

File: src/services/data-loader/extract-headings.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  if (Array.isArray(contentPart)) {
+    return contentPart.flatMap(extractHeadingContentPart);
+  }
+
+  return [];
+};
+
+export const extractHeadings = (content: Content): Heading[] => {
```
Comment (nlisgo) [inline src/services/data-loader/extract-headings.ts:27]: We should put some tests around this function.

File: src/model/mongodb/mongodb-repository.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ class MongoDBArticleRepository implements ArticleRepository {
   }
 
   async getArticleVersion(identifier: string): Promise<EnhancedArticleWithVersions> {
-    const allVersions = await this.versionedCollection.find({ $or: [{ _id: identifier }, { msid: identifier }] })
+    const allVersions = await this.versionedCollection.find({ msid: identifier })
```
Comment (nlisgo) [inline src/model/mongodb/mongodb-repository.ts:109]: If I understand this correctly, this is just clearing up. No need to use the `_id` identifier any more. If there are other things like this in this PR such as the removal of the test in `api.test.ts` then we should get them in a separate PR. There is enough to review here without the noise of unrelated changes.

File: package.json [hunk 3]
```diff
@@ -36,6 +36,7 @@
     "@aws-sdk/client-s3": "^3.279.0",
     "@aws-sdk/credential-providers": "^3.281.0",
     "@stencila/encoda": "0.121.1",
+    "@stencila/schema": "^1.18.0",
```
Comment (nlisgo) [inline package.json:39]: We should check with stencila that it's ok for us to use this for now. The archived the repo in January. 

https://github.com/stencila/schema#%EF%B8%8F-moved

File: src/services/data-loader/extract-headings.ts (hunk 1 above)
Comment (E-Kemp) [reply src/services/data-loader/extract-headings.ts:27]: Can do, it should be noted this was ripped straight from import to make typing a little easier. Perhaps a follow up ticket is needed to test, run and store the result of this in the database when the post request is made from import.

Comment (nlisgo) [reply src/services/data-loader/extract-headings.ts:27]: That's great context. The lack of tests was an issue over there too and happy for that to be a followup.

We should jump on a call with a few of us. There may be an opportunity to introduce this in a phased way so the likelihood of success in deployment is greater. If there are pieces like this that can be actioned in isolation then they are easier to approved then 3 PR's to 3 different repo's that need to be delivered together.

Comment (scottaubrey) [reply src/services/data-loader/extract-headings.ts:27]: It worth also pointing out that an almost identical version was the original basis for this function in import at https://github.com/elifesciences/enhanced-preprints-server/blob/87e27e254c4e1920be0437f3c501c24a4a2f0cf7/src/services/data-loader/data-loader.ts#L184-L216 . So it's kind of this functions grandparent, sat right next to it. :)

The thinking when we moved it to import was to make it so that epp-server didn't need to be aware of content schema at all - it just stores whatever it's given. That way we could remove the stencila dependencies in server and keep it "agnostic" of what is how those pieces are collected and transformed into these bits (import) and of how those are transformed into a final page (client). All it knows, is that client cares about a list of headers, separate to a (JSON) blob of content, separate to the abstract blob of content, etc. 

In fact - maybe it belongs in client after all?

Any way - we should work to resolve which is the latest/best/most-edge-cases-covered version, and make sure we use that wherever it lands.

File: src/model/mongodb/mongodb-repository.ts (hunk 2 above)
Comment (scottaubrey) [reply src/model/mongodb/mongodb-repository.ts:109]: Me and Elliot chatted about this yesterday. I think the conclusion was that perhaps it's not correct to remove this access pattern, but that it's probably not doing it right as was, and will need refining.

Separately, we want to remove returning all the content, and just return metadata, for the less interesting versions, but that's an easy follow up.

File: package.json (hunk 3 above)
Comment (scottaubrey) [reply package.json:39]: I reckon that's because they carry on work on the rust version that no longer depends on this repo. As I mentioned in another comment - I think we'd be better to work towards a stencila-schema-agnostic server if we can, but while we're still using the typescript version it should be fine.
